### PR TITLE
[CCUBE-1297][SR] Amended the toasts not moving up issue.

### PR DESCRIPTION
**Pull request by ShIrInRuTh**
**Changes**

-  Added onDismiss handler 
- Removing timeout and refs
- Tested the functionality for both toasts moving up and dismissal of toasts

**Additional information**

-   You may refer to this [CCUBE-1297](https://jira.ship.gov.sg/browse/CCUBE-1297)
- Fix for issue found from deskcheck

### CODE REVIEW

File: src/components/common/toast/toasts.tsx [hunk 1]
```diff
@@ -4,17 +4,13 @@ import { DisplayToast } from "./toast";
...
                 <DisplayToast
-                    key={value.id}
-                    toast={value}
-                    toastFunction={value.toastFunction}
+                    key={toast.id}
+                    toast={toast}
+                    toastFunction={toast.toastFunction}
                 />
             ))}
         </>
```
Comment (weili-govtech) [inline src/components/common/toast/toasts.tsx]:
```
    return toasts.map((toast) => (
        <DisplayToast
            key={toast.id}
            toast={toast}
            toastFunction={toast.toastFunction}
        />
    ));
```

File: src/components/common/toast/toasts.tsx [hunk 2]
```diff
@@ -4,17 +4,13 @@ import { DisplayToast } from "./toast";
...
         <>
-            {toasts.map((value) => (
+            {toasts.map((toast) => (
                 <DisplayToast
-                    key={value.id}
-                    toast={value}
-                    toastFunction={value.toastFunction}
+                    key={toast.id}
+                    toast={toast}
+                    toastFunction={toast.toastFunction}
```
Comment (weili-govtech) [inline src/components/common/toast/toasts.tsx]: apologies, on second thought, may I suggest to rename `toastFunction` as `onClickActionButton`?

apply this to `Toast` as well as `IToast`

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/common/toast/toasts.tsx (hunk 2 above)
Comment (ShIrInRuTh) [reply src/components/common/toast/toasts.tsx]: fixed in latest changes

File: src/components/common/toast/toasts.tsx (hunk 1 above)
Comment (ShIrInRuTh) [reply src/components/common/toast/toasts.tsx]: fixed in latest changes

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**